QualityControl/lib/dtos/KafkaConfigurationDto.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+ * See http://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+ * All rights not expressly granted are reserved.
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+import Joi from 'joi';
+
+/**
+ * @typedef {KafkaConfiguration}
+ * @param {boolean} enabled - if kafka should be used in current deployment
+ * @param {string} clientId - ID to be used as client and not qc-gui
+ * @param {Map<string, string>} consumerGroups - topics to be followed
+ * @param {string[]} brokers - brokers to connect to
+ */
+
+/**
+ * Function for validating the Kafka configuration
+ * @returns {KafkaConfiguration} - validated kafka configuration to be used
+ * @throws {InvalidInputError}
+ */
+export const KafkaConfigDto = Joi.object({
+  enabled: Joi.boolean().default(false),
+  clientId: Joi.string().custom((value, helpers) => {
+    if (process.env.NODE_ENV === 'PRODUCTION') {
+      return value;
+    }
+    if (value === 'qc-gui') {
+      return helpers.error('any.invalid');
```

Comment:
maybe it would be better to provide a more descriptive message so debugging is easier
```suggestion
      return helpers.error('any.invalid', { message: 'clientId "qc-gui" is not allowed outside PRODUCTION');
```

Reply:
Thank you for the suggestion. i believe your suggested change has a syntax error, missing `}`. Moreover, that message would not be shown as default.

I will update into using the message helper instead as per JOI docs:
https://joi.dev/api/?v=17.13.3

Reply:
in that case I would use `helper.error`

Reply:
`helpers.error` is to be used when we need a certain custom Joi error to be triggered further down the verification.
In our case, I believe `helpers.message` is the right way to go as it is the method that will be sending the custom message to our `try/catch`